packages/edit-site/src/hooks/commands/use-edit-mode-commands.js

```diff
@@ -158,7 +168,70 @@ function useManipulateDocumentCommands() {
 	};
 }
 
+function useAddNewPageCommand() {
+	const history = useHistory();
+	const { saveEntityRecord } = useDispatch( coreStore );
+	const { createErrorNotice } = useDispatch( noticesStore );
+
+	const createPageEntity = async ( { close } ) => {
+		try {
+			const _page = await saveEntityRecord(
+				'postType',
+				'page',
+				{
+					status: 'draft',
+				},
+				{
+					throwOnError: true,
+				}
+			);
+			if ( _page?.id ) {
+				history.push( {
+					postId: _page.id,
+					postType: 'page',
+					canvas: 'edit',
+				} );
+			}
+		} catch ( error ) {
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
+					? error.message
+					: __( 'An error occurred while creating the item.' );
+
+			createErrorNotice( errorMessage, {
+				type: 'snackbar',
+			} );
+		} finally {
+			close();
+		}
+	};
+
+	const commands = useMemo( () => {
+		return [
+			{
+				name: 'core/edit-site/add-new-page',
+				label: __( 'Add new page' ),
+				icon: plus,
+				callback: createPageEntity,
+			},
+		];
+	}, [ createPageEntity ] );
+
+	return {
+		isLoading: false,
+		commands,
+	};
+}
+
 export function useEditModeCommands() {
+	const { unregisterCommand } = useDispatch( commandsStore );
+	unregisterCommand( 'core/add-new-page' );
+
+	useCommandLoader( {
+		name: 'core/edit-site/add-new-page',
```

Comment:
I'm not sure, but should this re-use the same name as the core command, if we're unregistering the core one? Just wondering about anything that might programatically look at triggering existing commands by `name` 🤔

Reply:
Valid point.

I couldn't find anything in the [store](https://github.com/WordPress/gutenberg/tree/trunk/packages/commands/src/store) that enables triggering commands by name. 


I can change just to make sure.